satpy/resample.py

```diff
@@ -145,10 +145,12 @@
 from pyresample.geometry import SwathDefinition
 try:
     from pyresample.resampler import BaseResampler as PRBaseResampler
-    from pyresample.gradient import GradientSearchResampler
 except ImportError:
-    warnings.warn('Gradient search resampler not available, upgrade Pyresample.')
     PRBaseResampler = None
+try:
+    from pyresample.gradient import GradientSearchResampler
+except ImportError:
+    warnings.warn('Gradient search resampler not available. Maybe missing `shapely`?.')
```

Comment:
Is it possible to do this check inside some other function or to only show it when a user requests gradient search? It gets really annoying when most people probably aren't going to use the gradient search (for now at least).

Reply:
Moved the warning to a place where the user tries to use it.